src/inmanta/server/services/resourceservice.py

```diff
@@ -578,6 +579,15 @@ async def get_resource_actions(
                 f"The action_id parameter should be used in combination with either the first_timestamp or the last_timestamp "
                 f"Received action_id: {action_id}, first_timestamp: {first_timestamp}, last_timestamp: {last_timestamp}"
             )
+
+        if limit is None:
+            limit = APILIMIT
+        elif limit > APILIMIT:
+            raise BadRequest(
+                f"limit parameter can not exceed {APILIMIT}, got {limit}."
+                f" To retrieve more entries, use /api/v2/resource_actions"
```

Comment:
This is already the method handling `/api/v2/resource_actions`. Perhaps mention the pagination or the next/previous links in the response instead?

Reply:
I had considered that, my reasoning was the following

- when returning a 200 (or even a 301 or 302) with a next link, the paging is seamless for clients that support json-api paging protocol 
 - when returning an exception, it is clear to all clients that the response is no longer 'correct' (i.e. it is no longer backward compatible when returning large batches)
 - the first point ignores the fact that the other endpoint has a completely different return type, so the next page returns different types of objects

Reply:
Yes, that's true, I only meant that in this exception in this particular method, the user is already using the `/api/v2/resource_actions` endpoint, and in the message we could mention that pagination is possible using this endpoint, or just point to the docs, to check what to do, if they wish to retrieve more results.

Reply:
ah yes, copy past error on my behalf. Thank you.

Reply:
I changed this, but would it be beter to just silently return only 1000 records?

Reply:
the  jsonapi spec seems to allow this https://jsonapi.org/format/#fetching-pagination

Reply:
I think it's better this way